src/osd/PrimaryLogPG.cc

```diff
@@ -5703,6 +5792,70 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
       }
       break;
 
+    case CEPH_OSD_OP_SET_REDIRECT:
+      ++ctx->num_write;
+      {
+	if (pool.info.is_tier()) {
+	  result = -EINVAL;
+	  break;
+	}
+	if (!obs.exists) {
+	  result = -ENOENT;
+	  break;
+	}
```

Comment:
Oops, one more thing we need here:  if (!get_osdmap()->test_flag(CEPH_OSDMAP_REQIURE_LUMINOUS)).  Otherwise this might get set on an upgrading cluster that doesn't fully support/understand the redirects.